spec/nine_one_one_spec.rb

```diff
@@ -171,33 +98,109 @@
+        NineOneOne.configure do |config|
+          config.send_pagers = true
+          config.pager_duty_integration_key = 'integration key'
+          config.slack_enabled = true
+          config.slack_channel = 'channel'
+          config.slack_webhook_url = 'http://example.com'
+        end
+
+        notifier = NineOneOne.use(:default)
+
+        expect(notifier.notification_service).to be_an_instance_of(NineOneOne::SlackService)
+        expect(notifier.emergency_service).to be_an_instance_of(NineOneOne::PagerDutyService)
+      end
+    end
+
+    context 'custom config' do
+      it 'returns preconfigured notifier' do
+        NineOneOne.configure(:my_custom_config) do |config|
+          config.send_pagers = true
+          config.pager_duty_integration_key = 'integration key'
+          config.slack_enabled = true
+          config.slack_channel = 'channel'
+          config.slack_webhook_url = 'http://example.com'
+        end
+
+        notifier = NineOneOne.use(:my_custom_config)
+
+        expect(notifier.notification_service).to be_an_instance_of(NineOneOne::SlackService)
+        expect(notifier.emergency_service).to be_an_instance_of(NineOneOne::PagerDutyService)
+      end
+    end
   end
 
   describe '.emergency' do
-    let(:emergency_service) { instance_double(NineOneOne::PagerDutyService) }
+    let(:notifier) { instance_double(NineOneOne::Notifier) }
     let(:incident_key) { 'ERROR_KEY' }
     let(:message) { 'danger' }
     let(:details_hash) { { why: 'I dont know' } }
 
-    it 'delegates sending to emergency service' do
-      allow(NineOneOne).to receive(:emergency_service).and_return(emergency_service)
-
-      expect(emergency_service).to receive(:trigger_event).with(incident_key, message, details_hash)
+    it 'delegates sending to notifier' do
+      allow(NineOneOne).to receive(:use).and_return(notifier)
+      expect(notifier).to receive(:emergency).with(incident_key, message, details_hash)
 
       NineOneOne.emergency(incident_key, message, details_hash)
     end
   end
 
   describe '.notify' do
-    let(:notification_service) { instance_double(NineOneOne::SlackService) }
+    let(:notifier) { instance_double(NineOneOne::Notifier) }
     let(:message) { 'alert' }
 
-    it 'delegates notification to notification service' do
-      allow(NineOneOne).to receive(:notification_service).and_return(notification_service)
-
-      expect(notification_service).to receive(:notify).with(message)
+    it 'delegates notification to notifier' do
+      allow(NineOneOne).to receive(:use).with(:default).and_return(notifier)
+      expect(notifier).to receive(:notify).with(message)
 
       NineOneOne.notify(message)
     end
   end
+
+  describe '.notification_service' do
+    it 'returns default notification service' do
+      notifier = instance_double(NineOneOne::Notifier)
+      allow(NineOneOne).to receive(:use).with(:default).and_return(notifier)
+      expect(notifier).to receive(:notification_service)
```

Comment:
you might consider using Spies instead: https://relishapp.com/rspec/rspec-mocks/docs/basics/spies

(in general the latest rubocop cop (that's not used here) prefer it)